test/bigtable/filter.js

```diff
@@ -0,0 +1,677 @@
+      var str = Filter.convertToRegExpString(things);
+
+      assert.strictEqual(str, '(a|b|c)');
+    });
+
+    it('should not do anything to a string', function() {
+      var str1 = 'hello';
+      var str2 = Filter.convertToRegExpString(str1);
+
+      assert.strictEqual(str1, str2);
+    });
+
+    it('should convert a number to a string', function() {
+      var str = Filter.convertToRegExpString(1);
+
+      assert.strictEqual(str, '1');
+    });
+
+    it('should throw an error for unknown types', function() {
+      var errorMessage = /Can\'t convert to RegExp String from unknown type/;
+
+      assert.throws(function() {
+        Filter.convertToRegExpString(true);
+      }, errorMessage);
+    });
+  });
+
+  describe('createRange', function() {
+    it('should create a range object', function() {
+      var start = 'a';
+      var end = 'b';
+      var key = 'Key';
+
+      var range = Filter.createRange(start, end, key);
+
+      assert.deepEqual(range, {
+        startKeyInclusive: start,
+        endKeyInclusive: end
+      });
+    });
+
+    it('should only create start bound', function() {
+      var start = 'a';
+      var key = 'Key';
+
+      var range = Filter.createRange(start, null, key);
+
+      assert.deepEqual(range, {
+        startKeyInclusive: start
+      });
+    });
+
+    it('should only create an end bound', function() {
+      var end = 'b';
+      var key = 'Key';
+
+      var range = Filter.createRange(null, end, key);
+
+      assert.deepEqual(range, {
+        endKeyInclusive: end
+      });
+    });
+
+    it('should optionally accept inclusive flags', function() {
+      var start = {
+        value: 'a',
+        inclusive: false
+      };
+
+      var end = {
+        value: 'b',
+        inclusive: false
+      };
+
+      var key = 'Key';
+
+      var range = Filter.createRange(start, end, key);
+
+      assert.deepEqual(range, {
+        startKeyExclusive: start.value,
+        endKeyExclusive: end.value
+      });
+    });
+  });
+
+  describe('parse', function() {
+    it('should call each individual filter method', function() {
+      var row = Filter.prototype.row;
+      var value = Filter.prototype.value;
+
+      Filter.prototype.row = spy();
+      Filter.prototype.value = spy();
+
+      var fakeFilter = [{
+        row: 'a'
+      }, {
+        value: 'b'
+      }];
+
+      Filter.parse(fakeFilter);
+
+      assert.strictEqual(Filter.prototype.row.calls.length, 1);
+      assert.strictEqual(Filter.prototype.row.calls[0].args[0], 'a');
+      assert.strictEqual(Filter.prototype.value.calls.length, 1);
+      assert.strictEqual(Filter.prototype.value.calls[0].args[0], 'b');
+
+      Filter.prototype.row = row;
+      Filter.prototype.value = value;
+    });
+
+    it('should throw an error for unknown filters', function() {
+      var fakeFilter = [{
+        wat: 'a'
+      }];
+
+      assert.throws(
+        Filter.parse.bind(null, fakeFilter),
+        /Unknown filter\: wat/
```

Comment:
Like the earlier comment, please test `FilterError` separately, and then check that this method creates one.